sdk/cognitiveservices/azopenai/models.go

```diff
@@ -0,0 +1,209 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+// DO NOT EDIT.
+
+package azopenai
+
+// ClientGetCompletionsOptions contains the optional parameters for the Client.GetCompletions method.
+type ClientGetCompletionsOptions struct {
+	// placeholder for future optional parameters
+}
+
+// ClientGetEmbeddingsOptions contains the optional parameters for the Client.GetEmbeddings method.
+type ClientGetEmbeddingsOptions struct {
+	// placeholder for future optional parameters
+}
+
+type CompletionRequest struct {
+	// Generates bestof completions server-side and returns the "best" (the one with the highest log probability per token). Results
+	// cannot be streamed. When used with n, bestof controls the number of
+	// candidate completions and n specifies how many to return – bestof must be greater than n. Note: Because this parameter
+	// generates many completions, it can quickly consume your token quota. Use
+	// carefully and ensure that you have reasonable settings for maxtokens and stop. Has maximum value of 128.
+	BestOf *int32
+
+	// can be used to disable any server-side caching, 0=no cache, 1=prompt prefix enabled, 2=full cache
+	CacheLevel       *int32
+	CompletionConfig *string
+
+	// Echo back the prompt in addition to the completion
+	Echo *bool
+
+	// Number between -2.0 and 2.0. Positive values penalize new tokens based on their existing frequency in the text so far,
+	// decreasing the model's likelihood to repeat the same line verbatim.
+	FrequencyPenalty *float32
+
+	// Defaults to null. Modify the likelihood of specified tokens appearing in the completion. Accepts a json object that maps
+	// tokens (specified by their token ID in the GPT tokenizer) to an associated bias
+	// value from -100 to 100. You can use this tokenizer tool (which works for both GPT-2 and GPT-3) to convert text to token
+	// IDs. Mathematically, the bias is added to the logits generated by the model
+	// prior to sampling. The exact effect will vary per model, but values between -1 and 1 should decrease or increase likelihood
+	// of selection; values like -100 or 100 should result in a ban or exclusive
+	// selection of the relevant token. As an example, you can pass {"50256" : -100} to prevent the token from being generated.
+	LogitBias any
```

Comment:
I forgot that the code generator now has option `--rawjson-as-bytes` (false by default to avoid breaking changes).  This will expose any/any object fields as a `[]byte` instead of `any` or `map[string]any`.  The contents are raw JSON.  I believe that we want this to be on by default for new modules.

Reply:
This particular field though has a proper type now, @mikekistler must have fixed that awhile ago.

There aren't that many fields that are `any` so I'll need to check what they are supposed to be. 

I noticed that one of the `any` fields is a field that can either be a `string` or a `[]string`: [link](https://github.com/Azure/azure-sdk-for-go/pull/20617/files#diff-5dcbd3c85560a4917d6d5d8a85c0dadd7b251fb08c4eced4bf1cc7328eb531ffR400). We'll have to figure out the right way to represent this before we turn on --rawjson-as-bytes (which then turns it into a []byte). 

Filed issue: [link](https://github.com/Azure/azure-sdk-for-go/issues/21009)

Reply:
LogitBias type should be map[string]int